sematic/resolvers/state_machine_resolver.py

```diff
@@ -39,9 +40,13 @@ def resolve(self, future: AbstractFuture) -> typing.Any:
             if self._detach:
                 return self._detach_resolution(future)
 
+            signal.signal(signal.SIGINT, self._handle_sigint_cancel)
```

Comment:
we might want to wrap this in a try/except and then re-raise an appropriate error. This call can only happen on the main thread. If we see an exception, we will want to raise a reasonable error message, like `Resolvers must be invoked from the main thread, but was invoked on thread <name of current thread>`

Reply:
We might also want to check whether there is another handler already set to hand,e SIGINT, and wrap the existing handler so it still gets executed. Either that or raise an error that says you're not allowed to have such a signal handler present already.